src/shims/unix/fs.rs

```diff
@@ -900,14 +900,13 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         }
     }
 
-    fn linux_solarish_readdir64(
-        &mut self,
-        dirent_type: &str,
-        dirp_op: &OpTy<'tcx>,
-    ) -> InterpResult<'tcx, Scalar> {
+    fn readdir64(&mut self, dirent_type: &str, dirp_op: &OpTy<'tcx>) -> InterpResult<'tcx, Scalar> {
         let this = self.eval_context_mut();
 
-        if !matches!(&*this.tcx.sess.target.os, "linux" | "solaris" | "illumos") {
+        if !matches!(
+            &*this.tcx.sess.target.os,
+            "linux" | "solaris" | "illumos" | "macos" | "freebsd"
```

Comment:
Why "macos"? I don't think std uses this on macos. That means we have no tests, and we shouldn't add this without tests.